spec/requests/comments/updating_comment_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+require "support/comment_shared_contexts"
+
+RSpec.describe "Updating a comment", type: :request do
+  include_context "comment resource helpers"
+
+  context "when a different user tries to update" do
+    subject { create :comment }
+
+    it "fails when comment is not owned by user" do
+      put existing_comment_path,
+          {
+            resource_name: resource_name,
+            resource_id: resource_id,
+            id: Faker::Number.between(-5, 0),
+            content: Faker::Hacker.say_something_smart,
+            format: :json
+          }, authorization_header
+
+      expect(response.status).to eq 403
+    end
+  end
+
+  context "when same user tries to update" do
+    subject do
+      create :comment, user: valid_user
+    end
+
+    it "fails when comment text is empty" do
+      put existing_comment_path,
+          {
+            resource_name: resource_name,
+            resource_id: resource_id,
+            id: Faker::Number.between(-5, 0),
+            content: "",
+            format: :json
+          }, authorization_header
+
+      expect(response.status).to eq 400
+      expect(JSON.parse(response.body)["errors"]).to eq "Comment body can not be empty!"
+    end
+
+    it "updates successfully when all parameters are valid" do
+      new_text = Faker::Hacker.say_something_smart
+      put existing_comment_path,
+          {
+            resource_name: resource_name,
+            resource_id: resource_id,
+            id: subject.id,
```

Comment:
here you could do the `zhishi_url_option.merge({content: new_text, format: :json})` and for the others too.
Don't bother about the url_for_trackable, it would be a violation of OOP principles